roles/openshift_health_checker/openshift_checks/efk_stack_check.py

```diff
@@ -0,0 +1,257 @@
+            return es_status
+
+        # fix for 3.4 logging where es_master_name is getting set to an ip.
+        # so we set a try check around incase it fails just so it keeps working for 3.3
+        for pod in es_pods:
+            try:
+                if pod['status']['podIP'] == es_master_name:
+                    es_master_name = pod['metadata']['name']
+            except ValueError:
+                continue
+
+        # get cluster nodes
+        curl_cmd = "{} 'https://localhost:9200/_nodes'".format(self.es_curl)
+        node_cmd = "exec -ti {} -- {}".format(es_master_name, curl_cmd)
+
+        cluster_nodes = json.loads(self.exec_oc(node_cmd, task_vars))['nodes']
+        es_status['all_nodes_registered'] = 1
+        # The internal ES node name is a random string we do not track anywhere
+        # pylint: disable=unused-variable
+        for node, data in cluster_nodes.items():
+            has_matched = False
+            for pod in es_pods:
+                if data['host'] == pod['metadata']['name'] or data['host'] == pod['status']['podIP']:
+                    has_matched = True
+                    break
+
+            if has_matched is False:
+                es_status['all_nodes_registered'] = 0
+
+        return es_status
+
+    def check_es_cluster_health(self, es_pod, task_vars):
+        """Exec into the elasticsearch pod and check the cluster health"""
+        try:
+            curl_cmd = "{} 'https://localhost:9200/_cluster/health?pretty=true'".format(self.es_curl)
+            cluster_health = "exec -ti {} -- {}".format(es_pod, curl_cmd)
+            health_res = json.loads(self.exec_oc(cluster_health, task_vars))
+
+            if health_res['status'] == 'green':
+                return 2
+            elif health_res['status'] == 'yellow':
+                return 1
+            else:
+                return 0
+        except ValueError:
+            # The check failed so ES is in a bad state
+            return 0
+
+    def check_elasticsearch_diskspace(self, es_pod, task_vars):
+        """Exec into a elasticsearch pod and query the diskspace"""
+        results = {}
+        try:
+            disk_used = 0
+            disk_free = 0
+            trash_var = 0
+
+            disk_output = self.exec_oc("exec -ti {} -- df".format(es_pod), task_vars).split(' ')
+            disk_output = [x for x in disk_output if x]
+            for item in disk_output:
+                if item != "/elasticsearch/persistent":
+                    disk_used = disk_free
+                    disk_free = trash_var
+                    trash_var = item
+                else:
+                    break
+
+            results['used'] = int(disk_used)
+            results['free'] = int(disk_free)
+        except ValueError:
+            results['used'] = int(0)
+            results['free'] = int(0)
+
+        return results
+
+    def check_fluentd(self, fluentd_pods, task_vars):
+        """Verify fluentd is running"""
```

Comment:
We have some initial work on the logging repo to check fluentd status with 'how much it has read' and 'how much is left'.  This is to try to address 'logging records are slow to show in kibana' issues